pkg/controller/controller_instance.go

```diff
@@ -626,6 +630,16 @@ func (c *controller) reconcileServiceInstance(instance *v1beta1.ServiceInstance)
 		return err
 	}
 
+	// Check if the ServiceClass or ServicePlan has been deleted and do not allow
+	// new plans to be created. It's little more complicated since we do want to allow
+	// parameter changes on an instance that's been deleted.
```

Comment:
It's not the instance that's been deleted. I'd say: 

`It's a little complicated since we do want to allow parameter changes on an existing instance whose plan or class has been removed from the broker's catalog.`